csharp/src/Ice/UdpEndpoint.cs

```diff
@@ -4,61 +4,51 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Globalization;
 using System.Net;
 using System.Text;
 
 namespace ZeroC.Ice
 {
     /// <summary>The Endpoint class for the UDP transport.</summary>
-    public sealed class UdpEndpoint : IPEndpoint
+    internal sealed class UdpEndpoint : IPEndpoint
     {
         // We cannot move HashCompressFlag to IPEndpoint because of the way we marshal it in TcpEndpoint
         public override bool HasCompressionFlag { get; }
         public override bool IsDatagram => true;
 
-        /// <summary>The local network interface used to send multicast datagrams.</summary>
-        public string McastInterface { get; } = "";
-
-        /// <summary>The time-to-live of the multicast datagrams, in milliseconds.</summary>
-        public int McastTtl { get; } = -1;
+        public override string? this[string option] =>
+            option switch
+            {
+                "interface" => McastInterface,
+                "ttl" => _mcastTtl.ToString(CultureInfo.InvariantCulture),
+                _ => base[option],
+            };
 
-        public override int Timeout => -1;
         public override Transport Transport => Transport.UDP;
 
+        /// <summary>The local network interface used to send multicast datagrams.</summary>
+        internal string McastInterface { get; } = "";
+
         private readonly bool _connect;
         private int _hashCode = 0;
 
+        /// <summary>The time-to-live of the multicast datagrams, in hops.</summary>
+        private int _mcastTtl { get; } = -1;
```

Comment:
We should convert this to a TimeSpan at some point

Reply:
It's an integer, not a timespan.